### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by tianpf**

### CODE REVIEW

File: planar/megrez_test.go [hunk 1]
```diff
@@ -89,8 +89,8 @@ func TestAlgorithm_Equals(t *testing.T) {
 		want    bool
 		wantErr bool
 	}{
-		{name: "equals exact", args: args{g1: geometry1, g2: geometry3}, want: true, wantErr: false},
-		{name: "not equals exact", args: args{g1: geometry1, g2: geometry2}, want: false, wantErr: false},
+		{name: "equals exact", args: args{g1: geometry1, g2: geometry3, tolerance: 0.001}, want: true, wantErr: false},
```
Comment (zhyt1985) [inline planar/megrez_test.go]: `Equals` method don't need `tolerance` parameter.

Comment (tianpf) [reply planar/megrez_test.go]: take and done

File: algorithm/overlay/line_merge.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 //	or an empty atomic geometry, or an empty GEOMETRYCOLLECTION
 func LineMerge(ml matrix.Collection) matrix.Collection {
 	for i := 0; i < len(ml)-1; i++ {
-		for j := 1; j < len(ml); j++ {
+		for j := i + 1; j < len(ml); j++ {
```
Comment (zhyt1985) [inline algorithm/overlay/line_merge.go:13]: the length of list `ml` will change.

File: algorithm/calc/angle/quadrant.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package angle
+
+import (
+	"github.com/spatial-go/geoos/algorithm"
+	"github.com/spatial-go/geoos/algorithm/matrix"
+)
+
+// Quadrant four const
+const (
+	NE = iota + 1
```
Comment (zhyt1985) [inline algorithm/calc/angle/quadrant.go:10]: the values of `NE` and `NW` are same.

Comment (tianpf) [reply algorithm/calc/angle/quadrant.go:10]: take and done

File: algorithm/overlay/line_merge.go (hunk 2 above)
Comment (tianpf) [reply algorithm/overlay/line_merge.go:13]: take and done，
todo Rewrite with monotone chain

**Review by xiaogu-space — APPROVED** [review]

**Review by zhyt1985 — APPROVED** [review]

**Pull request merged**